frontend/src/container/PublicProfile.jsx

```diff
@@ -115,59 +120,129 @@ export class PublicProfile extends React.Component {
 
   onChangeAvatarClick = () => this.setState({ displayUploadPopup: POPUP_DISPLAY_STATE.AVATAR })
 
-  onChangeAvatarSuccess = () => {
+  onChangeAvatarSuccess = () => this.onChangeImageSuccess(
+    'avatar',
+    'profileAvatarImage',
+    'hasAvatar',
+    updateUserProfileAvatarName
+  )
+
+  onChangeCoverSuccess = () => this.onChangeImageSuccess(
+    'cover',
+    'profileCoverImage',
+    'hasCover',
+    updateUserProfileCoverName
+  )
+
+  onChangeImageSuccess = (basename, nameStateKey, hasImageKey, updateNameReducer) => {
     const { props, state } = this
-    const profileAvatarName = `avatar-${Date.now()}`
+    const name = `${basename}-${Date.now()}`
     if (state.displayedUser.userId === props.user.userId) {
-      this.props.dispatch(updateUserProfileAvatarName(profileAvatarName))
+      this.props.dispatch(updateNameReducer(name))
     }
     this.setState(oldState => {
       return {
         ...oldState,
-        displayedUser: { ...oldState.displayedUser, profileAvatarName, hasAvatar: true }
+        displayedUser: { ...oldState.displayedUser, nameStateKey: name, hasImageKey: true }
       }
     })
     this.onCloseUploadPopup()
   }
 
+  onChangeCoverClick = () => this.setState({ displayUploadPopup: POPUP_DISPLAY_STATE.COVER })
+
   onCloseUploadPopup = () => this.setState({ displayUploadPopup: undefined })
 
+  getPopupUploadImage = (uploadBaseUrl, onUploadSuccess, recommendedDimensions) => {
```

Comment:
I don't understand why you create a function that returns a component instead of putting the component directly in the render()

alternatively, you can create another component to avoid calling in the render() a function that returns a component

Reply:
It allows to share the common parameters without repeating them.
Moving this function as a component works with a small price: it is not immediately clear which props are needed. As a classic function the input parameters are explicit.

Reply:
Moved in a component (still in the same file as it is "private" and shares some constant definitions).